float64 fy
float64 fz
float64 old_fy
float64 old_fz
---
float64 x
float64 y
float64 ek
